egs/mgb2/asr/local/mgb_data_prep.sh

```diff
@@ -0,0 +1,186 @@
+  $dev_dir/wav_list
+
+for x in $(cat $dev_dir/wav_list); do 
+  echo $x $db_dir/dev/wav/$x.wav >> $dev_dir/wav.scp
+done
+
+
+# Creating a file reco2file_and_channel which is used by convert_ctm.pl in local/score.sh script
+awk '{print $1" "$1" A"}' $dev_dir/wav.scp > $dev_dir/reco2file_and_channel
+
+
+# Creating utt2spk for dev from segments
+if [ ! -f $dev_dir/utt2spk ]; then
+  cut -d ' ' -f1 $dev_dir/segments > $dev_dir/utt_id
+  cut -d '_' -f1-2 $dev_dir/utt_id | paste -d ' ' $dev_dir/utt_id - > $dev_dir/utt2spk
+fi
+if [ ! -f $dev_dir/spk2utt ]; then
+  utils/utt2spk_to_spk2utt.pl $dev_dir/utt2spk > $dev_dir/spk2utt
+fi
+
+# separate the overlapped dev files from non overlapped
+for list in overlap non_overlap; do
+  rm -rf ${dev_dir}_$list || true
+  cp -r $dev_dir ${dev_dir}_$list
+  for x in segments text utt2spk; do
+    utils/filter_scp.pl $db_dir/dev/${list}_speech $dev_dir/$x > ${dev_dir}_$list/${x}
+  done
+done
+
+# We are developing on non overapped data
+cp ${dev_dir}_non_overlap/* $dev_dir
+
+
+# Test
+cp $db_dir/test/text $test_dir/text
+cp $db_dir/test/segments.non_overlap_speech $test_dir/segments
+
+
+find $db_dir/test/wav -type f -name "*.wav" | \
+  awk -F/ '{print $NF}' | perl -pe 's/\.wav//g' > \
+  $test_dir/wav_list
+
+for x in $(cat $test_dir/wav_list); do 
+  echo $x $db_dir/test/wav/$x.wav >> $test_dir/wav.scp
+done
+
+# Creating a file reco2file_and_channel which is used by convert_ctm.pl in local/score.sh script
+awk '{print $1" "$1" A"}' $test_dir/wav.scp > $test_dir/reco2file_and_channel
+
+# Creating utt2spk for test from segments
+if [ ! -f $test_dir/utt2spk ]; then
+  cut -d ' ' -f1 $test_dir/segments > $test_dir/utt_id
+  cut -d '_' -f1-2 $test_dir/utt_id | paste -d ' ' $test_dir/utt_id - > $test_dir/utt2spk
+fi
+if [ ! -f $test_dir/spk2utt ]; then
+  utils/utt2spk_to_spk2utt.pl $test_dir/utt2spk > $test_dir/spk2utt
+fi
+
+# separate the overlapped test files from non overlapped
+for list in overlap non_overlap; do
+  rm -rf ${test_dir}_$list || true
+  cp -r $test_dir ${test_dir}_$list
+  for x in segments text utt2spk; do
+    utils/filter_scp.pl $db_dir/test/${list}_speech.lst $test_dir/$x > ${test_dir}_$list/${x}
+  done
+done
+
+# We are testing on non overapped data
+cp ${test_dir}_non_overlap/* $test_dir
+
+
+# Train
+find $db_dir/train/wav -type f -name "*.wav" | \
+  awk -F/ '{print $NF}' | perl -pe 's/\.wav//g' > \
+  $train_dir/wav_list
+
+for x in $(cat $train_dir/wav_list); do 
+  echo $x $db_dir/train/wav/$x.wav >> $train_dir/wav.scp
+done
+
+#Creating a file reco2file_and_channel which is used by convert_ctm.pl in local/score.sh script
+awk '{print $1" "$1" A"}' $train_dir/wav.scp > $train_dir/reco2file_and_channel
+
+# Creating utt2spk for train from segments
+#if [ ! -f $train_dir/utt2spk ]; then
+#  cut -d ' ' -f1 $train_dir/segments > $test_dir/utt_id
+#  cut -d '_' -f1-2 $train_dir/utt_id | paste -d ' ' $train_dir/utt_id - > $train_dir/utt2spk
+#fi
```

Comment:
You can remove these code if they are unnessasary

Reply:
done